File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(num)
    @num=num
  end

  def square_of_sum
    (1..@num).inject(0){ |sum,x| sum + x }**2
  end

  def sum_of_squares
    (1..@num).map{ |f| f**2 }.inject(0){ |sum,x| sum + x }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

const testVersion = 1

func SquareOfSums(num int) int {
	sum := float64(1+num) * (float64(num) / 2)
	return int(sum * sum)
}

func SumOfSquares(num int) int {
	counter := 0
	for i := 1; i <= num; i++ {
		counter += i * i
	}
	return counter
}

func Difference(num int) int {
	return SquareOfSums(num) - SumOfSquares(num)
}
